### ISSUE TITLE
@truffle/hdwallet-provider: ^1.2.0 not working with React-native 0.63.4

### ISSUE THREAD

**Issue by nikita-zot-1408** [trufflesuite/truffle#3917]
I am trying to implement @truffle/hdwallet-provider for importing user wallets with private key,mnemonic etc.

But the problem is I am always getting the same error over and over again:

 Error: PollingBlockTracker - encountered an error while attempting to update latest block

I implemented rn-nodeify in hope to fix the issue but I have no progress so far.

Any suggestions please?

**Comment by eggplantzzz** [comment]
@nikita-zot-1408 Interesting, sometimes that error occurs when it is having trouble connecting to the node. Are you using Infura?

**Comment by nikita-zot-1408** [comment]
No,I am not using infura. The thing is I used this library in the node.js and it worked perfectly. But when I moved it to the react-native it started giving this error

**Comment by eggplantzzz** [comment]
Is it possible that you could provide us with some code and reproduction steps so we can try and see if we can figure out why this is happening? Thanks!

**Comment by nikita-zot-1408** [comment]
@eggplantzzz Here is the function which is used for importing the wallet

```
export const importWalletKey = (privKey) => {
  const importKeyConstant = constants.PRIVATE_RPC;
  const provider = new HDWalletProvider(privKey, importKeyConstant, 0, 2);
  return provider;
};
```
and this is how I use it.

```
 const checkIfImported = await importWalletKey(`${privateKey}`);
 console.log(checkIfImported)      
```
privateKey is not an empty value I already checked it.

The react-native version is: 0.63.4 and the @truffle/hdwallet-provider version is:^1.2.0

**Comment by gnidan** [comment]
Hey @nikita-zot-1408 we'd love to help you sort this out. Could you make a minimal GH repo that shows this problem? We don't have a lot of expertise on the team with using react-native. Alternatively, if you're working off a private repository, could you add @eggplantzzz and me (@gnidan) so that we can more easily reproduce?

Thank you.
